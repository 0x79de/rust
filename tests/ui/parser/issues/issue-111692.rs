mod module {
    #[derive(Eq, PartialEq)]
    pub struct Type {
        pub x: u8,
        pub y: u8,
    }

    pub const C: u8 = 32u8;
}

fn test(x: module::Type) {
    if x == module::Type { x: module::C, y: 1 } { //~ ERROR struct literals are not allowed here
    }
}

fn test2(x: module::Type) {
    if x ==module::Type { x: module::C, y: 1 } { //~ ERROR struct literals are not allowed here
    }
}


fn test3(x: module::Type) {
    use module::Type;
    if x == Type { x: module::C, y: 1 } { //~ ERROR struct literals are not allowed here
    }
}

fn test4(x: module::Type) {
    use module as demo_module;
    if x == demo_module::Type { x: module::C, y: 1 } { //~ ERROR struct literals are not allowed here
    }
}

fn main() { }
