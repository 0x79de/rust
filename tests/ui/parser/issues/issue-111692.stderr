error: struct literals are not allowed here
  --> $DIR/issue-111692.rs:12:13
   |
LL |     if x == module::Type { x: module::C, y: 1 } {
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: surround the struct literal with parentheses
   |
LL |     if x == (module::Type { x: module::C, y: 1 }) {
   |             +                                   +

error: struct literals are not allowed here
  --> $DIR/issue-111692.rs:17:12
   |
LL |     if x ==module::Type { x: module::C, y: 1 } {
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: surround the struct literal with parentheses
   |
LL |     if x ==(module::Type { x: module::C, y: 1 }) {
   |            +                                   +

error: struct literals are not allowed here
  --> $DIR/issue-111692.rs:24:13
   |
LL |     if x == Type { x: module::C, y: 1 } {
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: surround the struct literal with parentheses
   |
LL |     if x == (Type { x: module::C, y: 1 }) {
   |             +                           +

error: struct literals are not allowed here
  --> $DIR/issue-111692.rs:30:13
   |
LL |     if x == demo_module::Type { x: module::C, y: 1 } {
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: surround the struct literal with parentheses
   |
LL |     if x == (demo_module::Type { x: module::C, y: 1 }) {
   |             +                                        +

error: aborting due to 4 previous errors

